src/System.Security.Permissions/ref/System.Security.Permissions.cs

```diff
@@ -190,7 +190,7 @@ public partial class DirectoryServicesPermissionEntry
     }
     public partial class DirectoryServicesPermissionEntryCollection : System.Collections.CollectionBase
     {
-        public DirectoryServicesPermissionEntryCollection() { }
+        internal DirectoryServicesPermissionEntryCollection() { }
```

Comment:
Internal members shouldn't be part of the ref assembly. If you remove them it will not try to add typeforwards to the facade and you will not see the errors you were getting

Reply:
Sure, I can remove the internal constructor. Though I wonder how come at least two occurrences below show internal ctor exposed in System.Security.Permissions 

https://github.com/dotnet/corefx/blob/4477d04229b93a69fd1733c8afcb7d3e15beb359/src/System.Security.Permissions/ref/System.Security.Permissions.cs#L294

and

https://github.com/dotnet/corefx/blob/4477d04229b93a69fd1733c8afcb7d3e15beb359/src/System.Security.Permissions/ref/System.Security.Permissions.cs#L294

Reply:
Making a constructor internal (or private) in the reference assembly source is permitted.  This prevents the compiler from adding the public default constructor.

Reply:
Ahh it makes sense. I didn't know that special constrain for constructors. then disregard my comment :smile: thanks @ericstj